glui.cpp

```diff
@@ -210,7 +264,17 @@ void GLUI_Main::setup_default_glut_callbacks( void )
   glutPassiveMotionFunc( glui_passive_motion_func );
   glutEntryFunc( glui_entry_func );
   glutVisibilityFunc( glui_visibility_func );
-  /*  glutIdleFunc( glui_idle_func );    // FIXME!  100% CPU usage!      */
+  /*  glutIdleFunc( glui_idle_func );    // FIXME!  100% CPU usage!      */  
+  //This is needed to host the X11 copy-paste selection-owner.
+  //It's really no big deal. The scheduler will divide time up
+  //and unless "vsync" is off a continuously rendering client
+  //will wait for the "vertical-blank."
+  //GLUT isn't a good fit for passive applications, and it is
+  //historically for full-motion 3-D displays.
+  //Clients can figure out a wait solution in their Idle function.
+  //freeglut provides a default idle, which it does not expose,
+  //but it's open-souce and so copy-friendly.
+  glutIdleFunc( glui_idle_func );
```

Comment:
One simple recommendation is to use the glutWindowStatusFunc (as recommended) and not redraw if offscreen (occluded or minimized, but oddly, not hidden) and in addition do a short sleep() of some kind in the idle when offscreen.

GLUI is C++. C++ programmers are supposed to be advanced. The elite. They have to learn the ropes somehow.